connectors/sync_job_runner.py

```diff
@@ -215,16 +216,23 @@ async def _sync_done(self, sync_status, sync_error=None):
     @with_concurrency_control()
     async def sync_starts(self):
         if not await self.reload_connector():
-            return False
+            raise SyncJobStartError(f"Couldn't reload connector {self.connector_id}")
 
         if self.connector.last_sync_status == JobStatus.IN_PROGRESS:
             logger.debug(
-                f"Connector {self.connector_id} is syncing, skip the job {self.job_id}..."
+                f"A sync job is started for connector {self.connector_id} by another connector instance, skipping..."
+            )
+            raise SyncJobStartError(
+                f"A sync job is started for connector {self.connector_id} by another connector instance"
             )
-            return False
 
-        await self.connector.sync_starts()
-        return True
+        try:
+            await self.connector.sync_starts()
+        except Exception as e:
+            if isinstance(e, elasticsearch.ConflictError):
+                raise
+            else:
+                raise SyncJobStartError(f"Unexpected error happened: {e}")
```

Comment:
I think it'll be more readable if you have multiple `except` instead of comparing with `isinstance`:

```
except ConflictError as e:
    raise
except Exception as e:
    raise SyncJobStartError(f"Unexpected error happened: {e}")

```

Reply:
> do you think second `raise` can be done with [exception context](https://docs.python.org/3/library/exceptions.html#exception-context)?

TIL